### ISSUE TITLE
Unable to override language grammars

### ISSUE THREAD

**Issue by Robot-Inventor** [expressive-code/expressive-code#214]
Thank you for the fantastic project. While using expressive-code in my Astro project, I noticed that I couldn't override some language grammars with custom ones.

I investigated the issue with the intention of sending a pull request and modified the code in plugin-shiki as follows, which now allows me to correctly override the language grammars with custom grammars. However, I encountered a type error. My skills were not sufficient to resolve this issue. Could you help me fix it?

https://github.com/expressive-code/expressive-code/blob/84cdd77f3d6c25ef1836af6cf7420fe9053116b6/packages/%40expressive-code/plugin-shiki/src/highlighter.ts#L27-L37

```ts
const langs: (ShikiLanguageInput | BundledLanguage)[] = [] 
if (config.langs?.length) {
	langs.push(...(config.langs as ShikiLanguageInput[]))
	const configLangs = config.langs.flat().map((lang) => lang.aliases?.length ? [lang.name, ...lang.aliases] : lang.name).flat()
	langs.push(...(Object.keys(bundledLanguages) as BundledLanguage[]).filter((lang) => !configLangs.includes(lang)))
} else {
	// For now, always preload all the languages because we had some strange race conditions
	// with lazy-loading that we couldn't solve yet. This is what the old version did as well.
	langs.push(...(Object.keys(bundledLanguages) as BundledLanguage[]))
}
highlighterPromise = getHighlighter({ 
	themes: [], 
	langs, 
})
```

**Comment by hippotastic** [comment]
Hello! Of course, happy to have a look later. 

In the meantime, could you elaborate on which languages you attempted to override without success? I'm asking because there are test cases in place which are supposed to validate this functionality: https://github.com/expressive-code/expressive-code/blob/main/packages/astro-expressive-code/test/fixtures/astro-4.0.0/ec.config.mjs

**Comment by Robot-Inventor** [comment]
Thank you for your quick response!

I attempted to load a custom language grammar for regexp from [this repository](https://github.com/Robot-Inventor/regex-syntax/blob/main/syntaxes/regex.tmLanguage.json). Unfortunately, I wasn't able to override the default grammar. Previously, regexp custom grammar used to be correctly loaded. However, since the [Shiki 1.6.0 release](https://github.com/shikijs/shiki/releases/tag/v1.6.0), where regexp was included as a default grammar, this no longer works.

On the other hand, I was able to successfully override the default grammar for shellsession with a custom language grammar from [this repository](https://github.com/Robot-Inventor/shell-session-syntax/blob/main/syntaxes/shell-session.tmLanguage.json). Also, if I don't use Expressive Code in my Astro project, I can override Shiki's regexp grammer.

To help illustrate the issue, I created a minimal repository that reproduces it: https://github.com/Robot-Inventor/ec-custom-grammar-issue . Please refer to it if needed.

**Comment by hippotastic** [comment]
Thank you for the reproduction, that was very helpful!

This should be now fixed in version 0.38.0, which I just released to NPM. I have rewritten the language loading logic so that custom languages provided to EC's `shiki.langs´ config option should no longer get overwritten by Shiki's internal operations. I also added a [new test case using your custom regexp grammar](https://github.com/expressive-code/expressive-code/blob/e16df97ec0fda0bd9c001425150320803806e03d/packages/%40expressive-code/plugin-shiki/test/rendering.test.ts#L415-L476) to help ensure there are no regressions in the future.
